src/object.d

```diff
@@ -3323,12 +3323,6 @@ unittest
         return _dup!(const(T), T)(a);
 }
 
-/// ditto
-@property T[] dup(T:void)(scope const(T)[] a) @trusted
-{
-    import core.internal.traits : Unconst;
-    return _dup!(const(T), Unconst!T)(a);
```

Comment:
This is more than just removing extra overloads. This fixes a safety issue.

I could compile the following code on the current compiler:

```D
void main() @safe
{
    immutable(int *)[] arr;
    arr ~= new int(4);
    const(void[]) b = arr;
    auto c = b.dup;
    pragma(msg, typeof(c)); // void[]
    int*[] arr2;
    arr2 ~= null;
    void[] x = arr2;
    x[] = c[];
    *arr2[0] = 5;
    assert(*arr[0] == 5);
}
```